Bug fixes

jonthegeek opened the pull request:
Closes several issues. I'm hoping github picks up the issues from the commits...

jonthegeek committed:
Deal with alternative spellings more completely.

Closes #36.

jonthegeek committed:
Dealt with contractions.

Also fixed a couple bugs along the way. And dealt with redirects.

jonthegeek committed:
Fix misspellings.

Closes #28.

jonthegeek committed:
Deal with deep links.

Closes #50.

jonthegeek committed:
vnum

jonthegeek committed:
More alt/spelling corner cases.

jonthegeek committed:
More alt/spelling corner cases.

jonthegeek committed:
Merge branch 'bug-fixes' of github.com:macmillancontentscience/wikimorphemes into bug-fixes

jonthegeek committed:
Correction for multiple alts/misspellings.

Also added a safety check for better exception reporting in .process_word_recursive.

jonthegeek committed:
Better example of multi-alt.

jonathanbratt commented on tests/testthat/test-word_splitters.R:
```diff
@@ -66,25 +66,25 @@ test_that(".split_inflections works", {
   # non-splittable words come back unchanged
```
```suggestion
  # non-splittable words come back as character(0)
```

jonathanbratt commented on tests/testthat/test-word_splitters.R:
```diff
@@ -66,25 +66,25 @@ test_that(".split_inflections works", {
+    character(0)
   )
 
   # non-English words come back unchanged, too
```
```suggestion
  # non-English words come back as character(0), too
```

jonathanbratt commented on tests/testthat/test-word_splitters.R:
```diff
@@ -66,25 +66,25 @@ test_that(".split_inflections works", {
+    character(0)
   )
 
   # irregular words come back unchanged
```
```suggestion
  # irregular words come back as character(0)
```

jonathanbratt reviewed (changes requested):
looks good! Just update some comments to match new conventions.

jonthegeek committed:
Slight clean-up based on full process.

This will not change the output, I'm just taking care of the 4 warnings I received this time (out of 1073055 words).

jonthegeek committed:
Fix corner cases.

Things identified by doing the full process.

jonthegeek committed:
Deal with template abuse.

Since we saw it in the wild.

jonathanbratt reviewed (approved):
yay!
